apps/explorer/src/core/helper/number-format-helper.ts

```diff
@@ -0,0 +1,502 @@
+     */
+    if (value.isGreaterThanOrEqualTo(TenThousand)) {
+      const gweiBN = new EthValue(value).toGweiBN().decimalPlaces(7)
+      return {
+        value: gweiBN.toFormat(),
+        unit: FormattedNumberUnit.GWEI,
+        tooltipText: `${new EthValue(value).toEthBN().toFixed()}`
+      }
+    }
+
+    /**
+     * Case IV: value < 10,000 wei
+     * Return: small values in WEI (no conversion) and tooltip with ETH value
+     */
+    return {
+      value: value.toFormat(),
+      unit: FormattedNumberUnit.WEI,
+      tooltipText: `${new EthValue(value).toEthBN().toFixed()}`
+    }
+  }
+
+  /**
+   * GROUP IV: Non-variable ETH values
+   * Convert a value in WEI to ETH
+   * @param value: BigNumber (in wei)
+   * @param allowRounding: whether or not to allow rounding
+   * @return FormattedNumber with value converted to ETH and tooltip if maxDecimalPlaces was applied
+   */
+  public static formatNonVariableEthValue(value: BigNumber, isException: boolean = false): FormattedNumber {
+    const ethBN = new EthValue(value).toEthBN()
+    const unit = FormattedNumberUnit.ETH
+    const dps = ethBN.decimalPlaces()
+
+    /**
+     * Case I: value === 0
+     * Return: "0 ETH"
+     */
+    if (ethBN.isZero()) {
+      return { value: '0', unit }
+    }
+
+    /**
+     * Case II: details page
+     * Return: return special case rounded number
+     */
+    if (isException) {
+      return this.getEthValueForDetailsPage(ethBN, dps)
+    }
+
+    /**
+     * Case III: x >= 1 Trillion
+     * Return: formated integer number with tooltip"
+     */
+    if (ethBN.isGreaterThanOrEqualTo(OneBillion)) {
+      return { ...this.formatIntegerValue(ethBN), unit: unit }
+    }
+
+    /**
+     * Case IV: x >= 1 Million
+     * Return: round number and tooltip if has decimal points
+     */
+    if (ethBN.isGreaterThanOrEqualTo(OneMillion)) {
+      return { ...this.getRoundNumber(ethBN, 0, dps), unit: unit }
+    }
+
+    /**
+     * Case V: x >= 1
+     * Return: round number to 2 decimal points and tooltip if > 2 decimal points
+     */
+    if (ethBN.isGreaterThanOrEqualTo(1)) {
+      return { ...this.getRoundNumber(ethBN, 2, dps), unit: unit }
+    }
+
+    /**
+     * Case VI: 0 <= x <= 0.0000001
+     * Return: round number upto 7 decimal points and tooltip if > 7 decimal points
+     */
+    if (ethBN.isGreaterThanOrEqualTo(SmallNumberBreakpoint)) {
+      return { ...this.getRoundNumber(ethBN, 7, dps), unit: unit }
+    }
+
+    /**
+     * Case V: value < 0.0000001
+     * Return: string "< 0.0000001" and tooltip with full value
+     */
+    return { value: '<0.0000001', unit, tooltipText: ethBN.toFormat() }
+  }
+
+  /**
+   * GROUP V: Non-variable GWei values
+   * Convert a value from WEI to GWEI
+   * @param value BigNumber (in wei)
+   * @return FormattedNumber with value in GWei and unit
+   */
+  public static formatNonVariableGWeiValue(value: BigNumber): FormattedNumber {
```

Comment:
We might want to change the name of this function as the units are now variable.